### ISSUE TITLE
Peformance vs hloc

### ISSUE THREAD

**Issue by FFFOCUS** [nianticlabs/ace#21]
1、I found it is low accuracy in low translation error vs hloc in my datasets， encoder net work down samples with 8，and that means every feature represent 8*8 grid，and target pixel position is the center of this grid，this position is not accuracy？
2、did you use unet network as encoder network for improving performance？and how can I train encoder network from scratch。very nice work and thanks a lot

**Comment by tcavallari** [comment]
Hello.

> encoder net work down samples with 8，and that means every feature represent 8*8 grid，and target pixel position is the center of this grid，this position is not accuracy？

Actually, it is true that the network downsamples by 8 but the convolutional nature of the network means that each feature is computed over a receptive field of ~80x80 pixels, so it has a very large view of the neighborhood around each input pixel. We saw comparable results between hLoc and ACE on the 7 and 12 Scenes datasets (see Table 1 of the paper), but I could see how other datasets might behave differently.

> did you use unet network as encoder network for improving performance

We didn't try that in our experiments, but I agree, it might improve performance by producing a higher resolution feature map to use in the MLP head training.

> how can I train encoder network from scratch?

We don't provide the encoder training code, but you can refer to Appendix A.1 of the [Arxiv paper](https://arxiv.org/abs/2305.14059) for the details of the encoder training process, and it should be possible to reimplement it from that.

**Comment by FFFOCUS** [comment]
thanks for your quickly reply！
I reopen this [Arxiv paper](https://arxiv.org/abs/2305.14059) again and find the Appendix and some experiment。
thanks again

**Comment by ebrach** [comment]
Regarding the performance vs. HLoc: There are multiple possibilities:
* As Tommaso points out, HLoc might just be genuinely better on your dataset
* If the scene is big, ACE might struggle to fit it all within a 4MB network. You can see whether [using an ACE ensemble ](https://github.com/nianticlabs/ace/issues/24#issuecomment-1641928728) improves the accuracy. 
* If your ground truth was generated using SfM, feature matching approaches (like HLoc) might have an advantage as discussed in this [ICCV21 paper](https://arxiv.org/abs/2109.00524).
